src/main/java/mate/academy/dao/impl/ManufacturerDaoJdbcImpl.java

```diff
@@ -0,0 +1,122 @@
+package mate.academy.dao.impl;
+
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.dao.ManufacturerDao;
+import mate.academy.exception.DataProcessingException;
+import mate.academy.lib.Dao;
+import mate.academy.model.Manufacturer;
+import mate.academy.util.ConnectionUtil;
+
+@Dao
+public class ManufacturerDaoJdbcImpl implements ManufacturerDao {
+
+    @Override
+    public Manufacturer create(Manufacturer manufacturer) {
+        String query = "INSERT INTO manufacturer.manufacturers (manufacturer_name, "
+                + "manufacturer_country) VALUES (?, ?);";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(query, Statement.RETURN_GENERATED_KEYS)) {
+            statement.setString(1, manufacturer.getName());
+            statement.setString(2, manufacturer.getCountry());
+            statement.executeUpdate();
+            ResultSet resultSet = statement.getGeneratedKeys();
+            while (resultSet.next()) {
+                manufacturer.setId(resultSet.getLong(1));
+            }
+            return manufacturer;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Something went wrong. Can't create manufacturer "
+                    + manufacturer, e);
+        }
+    }
+
+    @Override
+    public Optional<Manufacturer> getById(Long manufactureId) {
+        String query = "SELECT * FROM manufacturer.manufacturers "
+                + "WHERE manufacturer_id = ? AND manufacturer_delete = FALSE";
+        try (Connection connection = ConnectionUtil.getConnection(); PreparedStatement statement =
+                connection.prepareStatement(query)) {
+            statement.setLong(1, manufactureId);
+            ResultSet resultSet = statement.executeQuery();
+            if (resultSet.next()) {
+                Long id = resultSet.getLong("manufacturer_id");
+                String name = resultSet.getString("manufacturer_name");
+                String country = resultSet.getString("manufacturer_country");
+                Manufacturer manufacturer = new Manufacturer(name, country);
+                manufacturer.setId(id);
```

Comment:
let's move this logic into the private method `private Manufacturer parseDataFromResultSet(ResultSet resultSet) throws SQLException`

Reply:
Ok.